aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -0,0 +1,362 @@
+    lapackGetrf<scalar_t>(n, n, self_working_ptr, n, ipiv.data<int>(), &info);
+    infos[i] = info;
+    if (info != 0) {
+      return;
+    }
+
+    // Run twice, first to get the optimum work size
+    lwork = -1;
+    lapackGetri<scalar_t>(n, self_working_ptr, n, ipiv.data<int>(), &wkopt, lwork, &info);
+
+    lwork = static_cast<int>(wkopt);
+    work = at::empty({lwork}, self.type());
+
+    // now to compute the actual inverse
+    lapackGetri<scalar_t>(n, self_working_ptr, n, ipiv.data<int>(), work.data<scalar_t>(), lwork, &info);
+    infos[i] = info;
+    if (info != 0) {
+      return;
+    }
+  }
+}
+
+LINALG_HELPER_1_ARGS(inverse, self, cpu)
+
+Tensor inverse(const Tensor &self) {
+  if (self.size(-1) == 0) {
+    return at::empty_like(self);
+  }
+  if (self.dim() == 2) {
+    return at::_getri_single(self);
+  }
+  inverseCheckInputs(self);
+  return at::_inverse_helper(self);
+}
+
+Tensor& inverse_out(Tensor &result, const Tensor &self) {
+  if (self.size(-1) == 0) {
+    return result.resize_as_(self);
+  }
+  result.copy_(native::inverse(self));
+  return result;
+}
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ potrf ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template<typename scalar_t>
+static void apply_potrf(Tensor& A, bool upper) {
+#ifndef USE_LAPACK
+  AT_ERROR("not compiled with LAPACK");
+#endif
+
+  char uplo = upper ? 'U' : 'L';
+
+  auto A_data = A.data<scalar_t>();
+  auto A_mat_stride = matrixStride(A);
+
+  auto batch_size = batchCount(A);
+  int n = A.size(-2);
+  AT_CHECK(A.size(-1) == n, "last two dimensions must be of equal size");
+  //THArgCheck(THTensor_nDimensionLegacyAll(a) == 2, 1, "A should be 2 dimensional");
+  int lda = n;
+
+  for (int64_t i = 0; i < batch_size; i++) {
+    int info;
+    lapackPotrf(&uplo, &n, A_data+i*A_mat_stride, &lda, &info);
+    AT_CHECK(info == 0, "The leading minor of order ", info, " is not positive definite");
+  }
+}
+
+Tensor potrf_cpu(const Tensor &self, bool upper) {
+  if (self.dim() == 0) {
+    return self.sqrt();
+  } else if (self.size(-1) == 0) {
+    return at::empty_like(self);
+  }
+  AT_CHECK(self.dim() >= 2, "tensor must be at least two-dimensional");
+  Tensor result = cloneBatchedColumnMajor(self);
+  AT_DISPATCH_FLOATING_TYPES(result.type(), "potrf", [&] {
+      apply_potrf<scalar_t>(result, upper);
+    });
+  if (upper) {
+    result.triu_();
+  } else {
+    result.tril_();
+  }
+  return result;
+}
+
+Tensor& potrf_out(Tensor& result, const Tensor &self, bool upper) {
+  // should check if out is of the right format and copy before...
+  result.resize_as_(self).copy_(at::potrf(self));
+  return result;
+}
+
+template <typename scalar_t, bool inplace, bool upper>
+void apply_triu_tril(Tensor& result, const Tensor& self, int64_t k) {
+  auto n = self.size(-2);
+  auto m = self.size(-1);
+  auto self_batched_ = self.view({-1, n, m});
+  auto self_batched = self_batched_.accessor<scalar_t, 3>();
+  auto result_batched_ = result.view({-1, n, m});
```

Comment:
This doesn't work if `result` can't be viewed.
But we can't just replace it with `reshape`, because we wouldn't share storage in that case with `result`.
Should we care?

Reply:
Well, I've been thinking about this quite a bit in various contexts, but unless and until we introduce an iterator over multiple axes, I don't think we have good alternatives.
I think it's OK as an assumption, it'll handle any 3d matrices and quite a bit of more general ones. One might ask whether one might want a nicer error message, but I'm not sure that the cost/benefit is good.

Reply:
@t-vi @colesbury should be sending updates to TensorIterator today-ish that will add multiple-axes and reduction support

Reply:
Awesome! I can't say I'm understanding TensorIterator well enough yet (and particular not how to instantiate it...)